### DISCUSSION TITLE
evalf not giving the expect result

### DISCUSSION THREAD

**Discussion by miguelwon** [sympy/sympy#24745] [Q&A] [answered]
Hi,
Looking for some help related to `evalf`. I show bellow and example where `evalf` does not evaluate at required point: 

```
x = {i:Symbol(f'x{i}') for i in range(2)}
w = {i:Symbol(f'w{i}') for i in range(2)}
soma = sum([x[i] * w[i] for i in range(2)])
f = tanh(soma)
print(f)
```
gives:
```
tanh(w0*x0 + w1*x1)
```

Then:

```
df = diff(tanh(soma), x[1])
print(df)
```

gives:

```
w1*(1 - tanh(w0*x0 + w1*x1)**2)
```

and the evaluation:

```
df.evalf(subs={x[i]:0 for i in range(2)})
```

results in:
```
w1*(1 - tanh(w0*x0 + w1*x1)**2)
```

when it should result in  just`w1`.

**Comment by asmeurer** [comment, marked as answer]
This is a known issue with evalf(subs=...) https://github.com/sympy/sympy/issues/6974. It only works if you substitute every symbol. I would just use normal `subs`, especially if you are just going to substitute an exact value like 0.
